src/file.rs

```diff
@@ -21,12 +20,16 @@ pub struct Directory {
 }
 
 impl<'a> TryFrom<&'a Path> for Directory {
-    type Error = ActixError;
+    type Error = String;
     fn try_from(directory: &'a Path) -> Result<Self, Self::Error> {
-        let files = glob(directory.join("**").join("*").to_str().ok_or_else(|| {
-            error::ErrorInternalServerError("directory contains invalid characters")
-        })?)
-        .map_err(error::ErrorInternalServerError)?
+        let files = glob(
+            directory
+                .join("**")
+                .join("*")
+                .to_str()
+                .ok_or_else(|| String::from("directory contains invalid characters"))?,
+        )
+        .map_err(|e| e.msg)?
```

Comment:
What is the reason for this change?

Reply:
`ActixError` doesn't implement `Send`, so can't be moved across thread boundaries. I changed the error type to a `String`, and convert it to an `ActixError` in the view. This also has the benefit of making the error types more generic.